resources/ecr-scan-check.py

```diff
@@ -0,0 +1,204 @@
+        logoutput(
+            "Error", "No results found for image " + imagerepo + " with tag " + imagetag
+        )
+        sys.exit(1)
+
+    except ecr_client.exceptions.RepositoryNotFoundException as err:
+        logoutput("Error", "No repository found with name " + imagerepo)
+        sys.exit(1)
+
+    return response
+
+
+def parse_scan_results(response, severities):
+    logoutput("Info", "Severities included: " + ",".join(severities))
+    vulnerabilities = False
+    if len(response["imageScanFindings"]["findingSeverityCounts"]) > 0:
+        for severity in response["imageScanFindings"]["findingSeverityCounts"]:
+            if severity in severities:
+                vulnerabilities = True
+                count = response["imageScanFindings"]["findingSeverityCounts"][severity]
+                logoutput("Error", "{}: {}".format(severity, count))
+
+    return vulnerabilities
+
+
+def generate_report_url(ecr_scanfindings, region):
+    report_url = "https://{region}.console.aws.amazon.com/ecr/repositories/private/{registry}/{repository}/image/{digest}/scan-results/?region={region}".format(
+        region=region,
+        registry=ecr_scanfindings["registryId"],
+        repository=ecr_scanfindings["repositoryName"],
+        digest=ecr_scanfindings["imageId"]["imageDigest"],
+    )
+
+    return report_url
+
+
+def generate_slack_json(ecr_imagedata, vulnerabilities, report_url):
+    if vulnerabilities:
+        colour = "#FF0000"
+        heading = "ECR vulnerability scan failure"
+        result = "Vulnerabilities found"
+    else:
+        colour = "#00FF00"
+        heading = "Successful ECR vulnerability scan"
+        result = "No vulnerabilities"
+
+    image_name = ecr_imagedata["repositoryName"]
+    image_tag = ecr_imagedata["imageId"]["imageTag"]
+
+    slack_json = [
+        {
+            "fallback": heading,
+            "pretext": heading
+            + " for *<$ATC_EXTERNAL_URL/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|$BUILD_PIPELINE_NAME>*",
+            "fields": [
+                {"title": "Image Name", "value": image_name, "short": "true"},
+                {"title": "Image Tag", "value": image_tag, "short": "true"},
+                {
+                    "title": "Scan Result:",
+                    "value": "<" + report_url + "|" + result + ">",
+                    "short": "false",
+                },
+            ],
+            "color": colour,
+            "footer": "$ATC_EXTERNAL_URL",
+        }
+    ]
+
+    report_dir = "scan-report"
+    if not os.path.exists(report_dir):
+        os.mkdir(report_dir)
+
+    json_path = "./" + report_dir + "/report.json"
+    with open(json_path, "w", encoding="utf-8") as result_file:
+        json.dump(slack_json, result_file, ensure_ascii=False, indent=4)
+
+    result_file.close()
+
+
+if __name__ == "__main__":
+    # Ensure required environment vars are set
```

Comment:
I'd pull this out into a separate function like [this](https://github.com/companieshouse/ois-tuxedo-qsp-transfer/blob/main/qsp_transfer.py#L18). It also has the added benefit of giving details of every variable that's missing in one hit rather than potentially having to hit each in turn